sumologic/resource_sumologic_muting_schedules_library_muting_schedule.go

```diff
@@ -0,0 +1,324 @@
+package sumologic
+
+import (
+	"log"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceSumologicMutingSchedulesLibraryMutingSchedule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceSumologicMutingSchedulesLibraryMutingScheduleCreate,
+		Read:   resourceSumologicMutingSchedulesLibraryMutingScheduleRead,
+		Update: resourceSumologicMutingSchedulesLibraryMutingScheduleUpdate,
+		Delete: resourceSumologicMutingSchedulesLibraryMutingScheduleDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: getMutingScheduleSchema(),
+	}
+}
+
+func getMutingScheduleBaseSchema() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+
+		"name": {
+			Type:     schema.TypeString,
+			Required: true,
+			ValidateFunc: validation.All(
+				validation.StringDoesNotContainAny("/"),
+				validation.StringLenBetween(1, 255),
+				validation.StringMatch(regexp.MustCompile(`(?s)^[^\ ].*[^\ ]$`),
+					"name must not contain leading or trailing spaces"),
+			),
+		},
+
+		"description": {
+			Type:     schema.TypeString,
+			Optional: true,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(0, 4096),
+				validation.StringMatch(regexp.MustCompile(`(?s)^[^\ ].*[^\ ]$`),
+					"description must not contain leading or trailing spaces"),
+			),
+		},
+
+		"parent_id": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Computed: true,
+		},
+
+		"monitor": {
+			Type:     schema.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: getMonitorScopeSchemma(),
+			},
+		},
+
+		"schedule": {
+			Type:     schema.TypeList,
+			Required: true,
+			MaxItems: 1,
+			Elem: &schema.Resource{
+				Schema: getScheduleDefinitionSchemma(),
+			},
+		},
+
+		"version": {
+			Type:     schema.TypeInt,
+			Optional: true,
+			Computed: true,
+		},
+	}
+}
+
+func getMonitorScopeSchemma() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"ids": {
+			Type:     schema.TypeList,
+			Optional: true,
+			Elem: &schema.Schema{
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+		"all": {
+			Type:     schema.TypeBool,
+			Optional: true,
+		},
+	}
+}
+
+func getScheduleDefinitionSchemma() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"timezone": {
+			Type:     schema.TypeString,
+			Required: true,
+			ForceNew: false,
```

Comment:
"timezone" is the kind of field that won't need to change for every update or it could be a property or configuration value from the customer TF code. 

that is the reason I put ForceNew: false. 

And I based on the definition of the following. 
`In Terraform, the ForceNew attribute is used to control whether a resource should be replaced (destroyed and re-created) or updated when its definition changes. Setting ForceNew=false means that Terraform will attempt to update the existing resource in place, keeping its existing state, rather than requiring replacement.`

I am not an expert in TF. My understanding could be wrong. And welcome the right suggestion.

Reply:
I'm neither but i was curious. @sumovishal can u help here. Thanks